// csgo.js/src/lib/dictionary.js
/* name the stats (ant-like work) */

const prettyTime = (seconds) => {
    const h = Math.floor(seconds / 3600);
    const m = Math.floor((seconds % 3600) / 60);
    return `${h}h ${m}m`
}

module.exports = {
    total_kills: {
        name: 'Kills',
        key: 'kills',
        category: 'Stats',
        transform: (value) => value,
    },
    total_deaths: {
        name: 'Deaths',
        key: 'deaths',
        category: 'Stats',
        transform: (value) => value,
    },
    total_time_played: {
        name: 'Time played',
        key: 'time_played',
        category: 'Stats',
        transform: (value) => ({
            seconds: value,
            hours: Math.floor(value / 3600),
            display: prettyTime(value)
        }),
    },
    total_planted_bombs: {
        name: 'Bombs Planted',
        key: 'planted_bombs',
        category: 'Stats',
        transform: (value) => value,
    },
    total_defused_bombs: {
        name: 'Bombs Defused',
        key: 'defused_bombs',
        category: 'Stats',
        transform: (value) => value,
    },
    total_wins: {
        name: 'Wins',
        key: 'wins',
        category: 'Stats',
        transform: (value) => value,
    },
    total_damage_done: {
        name: 'Damage Done',
        key: 'damage_done',
        category: 'Stats',
        transform: (value) => value,
    },
    total_money_earned: {
        name: 'Money Earned',
        key: 'money_earned',
        category: 'Stats',
        transform: (value) => value,
    },
    total_kills_knife: {
        name: 'Knife Kills',
        key: 'knife_kills',
        category: 'Stats',
        transform: (value) => value,
    },
    total_kills_hegrenade: {
        name: 'HE Grenade Kills',
        key: 'hegrenade_kills',
        category: 'Stats',
        transform: (value) => value,
    },
    total_kills_glock: {
        name: 'Glock Kills',
        key: 'glock_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_deagle: {
        name: 'Deagle Kills',
        key: 'deagle_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_elite: {
        name: 'Elite Kills',
        key: 'elite_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_fiveseven: {
        name: 'Fiveseven Kills',
        key: 'fiveseven_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_xm1014: {
        name: 'XM 1014 Kills',
        key: 'xm1014_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_mac10: {
        name: 'Mac10 Kills',
        key: 'mac10_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_ump45: {
        name: 'UMP45 Kills',
        key: 'ump45_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_p90: {
        name: 'P90 Kills',
        key: 'p90_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_awp: {
        name: 'AWP Kills',
        key: 'awp_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_ak47: {
        name: 'AK47 Kills',
        key: 'ak47_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_aug: {
        name: 'AUG Kills',
        key: 'aug_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_famas: {
        name: 'FAMAS Kills',
        key: 'famas_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_g3sg1: {
        name: 'G3SG1 Kills',
        key: 'g3sg1_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_m249: {
        name: 'M249 Kills',
        key: 'm249_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_headshot: {
        name: 'Headshot Kills',
        key: 'headshot_kills',
        category: 'Stats',
        transform: (value) => value,
    },
    total_kills_enemy_weapon: {
        name: 'Kills with enemy weapons',
        key: 'enemy_weapon_kills',
        category: 'Stats',
        transform: (value) => value,
    },
    total_wins_pistolround: {
        name: 'Pistol rounds won',
        key: 'pistol_round_wins',
        category: 'Stats',
        transform: (value) => value,
    },
    total_wins_map_cs_assault: {
        name: 'Rounds won in Assault',
        key: 'cs_assault_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_cs_italy: {
        name: 'Rounds won in Italy',
        key: 'cs_italy_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_cs_office: {
        name: 'Rounds won in Office',
        key: 'cs_office_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_cbble: {
        name: 'Rounds won in Cobblestone',
        key: 'de_cbble_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_dust2: {
        name: 'Rounds won in Dust II',
        key: 'de_dust2_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_dust: {
        name: 'Rounds won in Dust',
        key: 'de_dust_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_inferno: {
        name: 'Rounds won in Inferno',
        key: 'de_inferno_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_nuke: {
        name: 'Rounds won in Nuke',
        key: 'de_nuke_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_train: {
        name: 'Rounds won in Train',
        key: 'de_train_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_weapons_donated: {
        name: 'Weapons Donated',
        key: 'weapons_donated',
        category: 'Stats',
        transform: (value) => value,
    },
    total_broken_windows: {
        name: 'Windows Broken',
        key: 'broken_windows',
        category: 'Stats',
        transform: (value) => value,
    },
    total_kills_enemy_blinded: {
        name: 'Blinded Enemy Kills',
        key: 'kills_enemy_blinded',
        category: 'Stats',
        transform: (value) => value,
    },
    total_kills_knife_fight: {
        name: 'Kills in Knife Fight',
        key: 'kills_knife_fight',
        category: 'Stats',
        transform: (value) => value,
    },
    total_kills_against_zoomed_sniper: {
        name: 'Zoomed Sniper Kills',
        key: 'kills_against_zoomed_sniper',
        category: 'Stats',
        transform: (value) => value,
    },
    total_dominations: {
        name: 'Dominations',
        key: 'dominations',
        category: 'Stats',
        transform: (value) => value,
    },
    total_domination_overkills: {
        name: 'Dominations overkills',
        key: 'domination_overkills',
        category: 'Stats',
        transform: (value) => value,
    },
    total_revenges: {
        name: 'Revenges',
        key: 'revenges',
        category: 'Stats',
        transform: (value) => value,
    },
    total_shots_hit: {
        name: 'Shots Hit',
        key: 'shots_hit',
        category: 'Stats',
        transform: (value) => value,
    },
    total_shots_fired: {
        name: 'Shots fired',
        key: 'shots_fired',
        category: 'Stats',
        transform: (value) => value,
    },
    total_rounds_played: {
        name: 'Rounds Played',
        key: 'rounds_played',
        category: 'Stats',
        transform: (value) => value,
    },
    total_shots_deagle: {
        name: 'Deagle shots',
        key: 'deagle_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_glock: {
        name: 'Glock shots',
        key: 'glock_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_elite: {
        name: 'Elite Shots',
        key: 'elite_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_fiveseven: {
        name: 'Five Seven Shots',
        key: 'fiveseven_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_awp: {
        name: 'AWP Shots',
        key: 'awp_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_ak47: {
        name: 'AK47 Shots',
        key: 'ak47_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_aug: {
        name: 'AUG Shots',
        key: 'aug_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_famas: {
        name: 'Famas Shots',
        key: 'famas_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_g3sg1: {
        name: 'G3SG1 Shots',
        key: 'g3sg1_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_p90: {
        name: 'P90 Shots',
        key: 'p90_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_mac10: {
        name: 'Mac10 Shots',
        key: 'mac10_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_ump45: {
        name: 'UMP45 Shots',
        key: 'ump45_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_xm1014: {
        name: 'XM1014 Shots',
        key: 'xm1014_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_m249: {
        name: 'M249 Shots',
        key: 'm249_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_deagle: {
        name: 'Deagle Hits',
        key: 'deagle_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_glock: {
        name: 'Glock Hits',
        key: 'glock_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_elite: {
        name: 'Elite Hits',
        key: 'elite_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_fiveseven: {
        name: 'Five Seven Hits',
        key: 'fiveseven_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_awp: {
        name: 'AWP Hits',
        key: 'awp_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_ak47: {
        name: 'AK47 Hits',
        key: 'ak47_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_aug: {
        name: 'AUG Hits',
        key: 'aug_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_famas: {
        name: 'Famas Hits',
        key: 'famas_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_g3sg1: {
        name: 'G3SG1 Hits',
        key: 'g3sg1_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_p90: {
        name: 'P90 Hits',
        key: 'p90_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_mac10: {
        name: 'Mac10 Hits',
        key: 'mac10_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_ump45: {
        name: 'UMP45 Hits',
        key: 'ump45_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_xm1014: {
        name: 'XM1014 Hits',
        key: 'xm1014_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_m249: {
        name: 'M249 Hits',
        key: 'm249_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_rounds_map_cs_assault: {
        name: 'Rounds played in Assault',
        key: 'cs_assault_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_cs_italy: {
        name: 'Rounds played in Italy',
        key: 'cs_italy_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_cs_office: {
        name: 'Rounds played in Office',
        key: 'cs_office_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_de_cbble: {
        name: 'Rounds played in Cobblestone',
        key: 'de_cbble_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_de_dust2: {
        name: 'Rounds played in Dust II',
        key: 'de_dust2_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_de_inferno: {
        name: 'Rounds played in Inferno',
        key: 'de_inferno_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_de_nuke: {
        name: 'Rounds played in Nuke',
        key: 'de_nuke_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_de_train: {
        name: 'Rounds played in Train',
        key: 'de_train_played',
        category: 'Maps',
        transform: (value) => value,
    },
    last_match_t_wins: {
        name: 'LastMatch Rounds wins as Terrorist',
        key: 'last_match_t_wins',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_ct_wins: {
        name: 'LastMatch Rounds wins as Counter-Terrorist',
        key: 'last_match_ct_wins',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_wins: {
        name: 'LastMatch rounds won',
        key: 'last_match_wins',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_max_players: {
        name: 'LastMatch Max Players',
        key: 'last_match_max_players',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_kills: {
        name: 'LastMatch Kills',
        key: 'last_match_kills',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_deaths: {
        name: 'LastMatch Deaths',
        key: 'last_match_deaths',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_mvps: {
        name: 'LastMatch MVPs',
        key: 'last_match_mvps',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_favweapon_id: {
        name: 'LastMatch Favorite Weapon ID',
        key: 'last_match_favweapon_id',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_favweapon_shots: {
        name: 'LastMatch Favorite Weapon shots',
        key: 'last_match_favweapon_shots',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_favweapon_hits: {
        name: 'LastMatch Favorite Weapon hits',
        key: 'last_match_favweapon_hits',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_favweapon_kills: {
        name: 'LastMatch Favorite Weapon kills',
        key: 'last_match_favweapon_kills',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_damage: {
        name: 'LastMatch Damage',
        key: 'last_match_damage',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_money_spent: {
        name: 'LastMatch Money Spent',
        key: 'last_match_money_spent',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_dominations: {
        name: 'LastMatch Dominations',
        key: 'last_match_dominations',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_revenges: {
        name: 'LastMatch Revenges',
        key: 'last_match_revenges',
        category: 'LastMatch',
        transform: (value) => value,
    },
    total_mvps: {
        name: 'MVPs',
        key: 'mvps',
        category: 'Stats',
        transform: (value) => value,
    },
    total_rounds_map_de_lake: {
        name: 'Rounds played in Lake',
        key: 'de_lake_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_de_safehouse: {
        name: 'Rounds played in Safehouse',
        key: 'de_safehouse_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_de_stmarc: {
        name: 'Rounds played in St Marc',
        key: 'de_stmarc_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_de_bank: {
        name: 'Rounds played in Bank',
        key: 'de_bank_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_de_shorttrain: {
        name: 'Rounds played in Train (wingman)',
        key: 'de_shorttrain_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_TR_planted_bombs: {
        name: 'Kills',
        key: 'TR_planted_bombs',
        category: 'Unknown',
        transform: (value) => value,
    },
    total_gun_game_rounds_won: {
        name: 'gun_game_rounds_won',
        key: 'gun_game_rounds_won',
        category: 'Unknown',
        transform: (value) => value,
    },
    total_gun_game_rounds_played: {
        name: 'gun_game_rounds_played',
        key: 'gun_game_rounds_played',
        category: 'Unknown',
        transform: (value) => value,
    },
    total_wins_map_de_house: {
        name: 'Rounds won in House',
        key: 'de_house_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_vertigo: {
        name: 'Rounds won in Vertigo',
        key: 'de_vertigo_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_ar_monastery: {
        name: 'Rounds played in Monastery',
        key: 'ar_monastery_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_ar_shoots: {
        name: 'Rounds played in Shoots',
        key: 'ar_shoots_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_rounds_map_ar_baggage: {
        name: 'Rounds played in Baggage',
        key: 'ar_baggage_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_ar_shoots: {
        name: 'Rounds won in Shoots',
        key: 'ar_shoots_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_ar_baggage: {
        name: 'Rounds won in Baggage',
        key: 'ar_baggage_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_lake: {
        name: 'Rounds won in Lake',
        key: 'de_lake_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_stmarc: {
        name: 'Rounds won in St Marc',
        key: 'de_stmarc_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_shorttrain: {
        name: 'Rounds won in Train (wingman)',
        key: 'de_shorttrain_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_wins_map_de_safehouse: {
        name: 'Rounds won in Safehouse',
        key: 'de_safehouse_wins',
        category: 'Maps',
        transform: (value) => value,
    },
    total_matches_won: {
        name: 'Matches Won',
        key: 'matches_won',
        category: 'Stats',
        transform: (value) => value,
    },
    total_matches_played: {
        name: 'Matches Played',
        key: 'matches_played',
        category: 'Stats',
        transform: (value) => value,
    },
    total_gg_matches_won: {
        name: 'GG Matches Won',
        key: 'gg_matches_won',
        category: 'Unknown',
        transform: (value) => value,
    },
    total_gg_matches_played: {
        name: 'GG Matches Played',
        key: 'gg_matches_played',
        category: 'Unknown',
        transform: (value) => value,
    },
    total_progressive_matches_won: {
        name: 'progressive_matches_won',
        key: 'progressive_matches_won',
        category: 'Unknown',
        transform: (value) => value,
    },
    total_trbomb_matches_won: {
        name: 'trbomb_matches_won',
        key: 'trbomb_matches_won',
        category: 'Unknown',
        transform: (value) => value,
    },
    total_contribution_score: {
        name: 'Contribution Score',
        key: 'contribution_score',
        category: 'Stats',
        transform: (value) => value,
    },
    last_match_contribution_score: {
        name: 'LastMatch Contribution Score',
        key: 'last_match_contribution_score',
        category: 'LastMatch',
        transform: (value) => value,
    },
    last_match_rounds: {
        name: 'LastMatch Rounds Played',
        key: 'last_match_rounds',
        category: 'LastMatch',
        transform: (value) => value,
    },
    total_kills_hkp2000: {
        name: 'USP-S Kills',
        key: 'hkp2000_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_hkp2000: {
        name: 'USP-S Shots',
        key: 'hkp2000_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_hkp2000: {
        name: 'USP-S Hits',
        key: 'hkp2000_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_p250: {
        name: 'P250 Hits',
        key: 'p250_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_p250: {
        name: 'P250 Kills',
        key: 'p250_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_p250: {
        name: 'P250 Shots',
        key: 'p250_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_sg556: {
        name: 'SG556 Kills',
        key: 'sg556_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_sg556: {
        name: 'SG556 Shots',
        key: 'sg556_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_sg556: {
        name: 'SSG 556',
        key: 'sg556_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_scar20: {
        name: 'SCAR20 Kills',
        key: 'scar20_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_scar20: {
        name: 'SCAR20 Kills',
        key: 'scar20_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_scar20: {
        name: 'SCAR20 Shots',
        key: 'scar20_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_ssg08: {
        name: 'SSG08 Shots',
        key: 'ssg08_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_ssg08: {
        name: 'SSG08 Hits',
        key: 'ssg08_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_ssg08: {
        name: 'SSG08 Kills',
        key: 'ssg08_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_mp7: {
        name: 'MP7 Shots',
        key: 'mp7_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_mp7: {
        name: 'MP7 Hits',
        key: 'mp7_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_mp7: {
        name: 'MP7 Kills',
        key: 'mp7_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_mp9: {
        name: 'MP9 Kills',
        key: 'mp9_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_mp9: {
        name: 'MP9 Shots',
        key: 'mp9_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_mp9: {
        name: 'MP9 Hits',
        key: 'mp9_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_nova: {
        name: 'Nova Hits',
        key: 'nova_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_nova: {
        name: 'Nova Kills',
        key: 'nova_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_nova: {
        name: 'Nova Shots',
        key: 'nova_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_negev: {
        name: 'Negev hits',
        key: 'negev_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_negev: {
        name: 'Negev Kills',
        key: 'negev_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_negev: {
        name: 'Negev Shots',
        key: 'negev_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_sawedoff: {
        name: 'SawedOff Shots',
        key: 'sawedoff_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_sawedoff: {
        name: 'SawedOff Hits',
        key: 'sawedoff_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_sawedoff: {
        name: 'SawedOff Kills',
        key: 'sawedoff_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_bizon: {
        name: 'Bizon Shots',
        key: 'bizon_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_bizon: {
        name: 'Bizon Hits',
        key: 'bizon_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_bizon: {
        name: 'Bizon Kills',
        key: 'bizon_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_tec9: {
        name: 'Tec9 Kills',
        key: 'tec9_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_tec9: {
        name: 'Tec9 Shots',
        key: 'tec9_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_tec9: {
        name: 'Tec9 Hits',
        key: 'tec9_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_mag7: {
        name: 'Mag7 Shots',
        key: 'mag7_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_mag7: {
        name: 'Mag7 Hits',
        key: 'mag7_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_mag7: {
        name: 'Mag7 Kills',
        key: 'mag7_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_gun_game_contribution_score: {
        name: 'gun_game_contribution_score',
        key: 'gun_game_contribution_score',
        category: 'Unknown',
        transform: (value) => value,
    },
    last_match_gg_contribution_score: {
        name: 'last_match_gg_contribution_score',
        key: 'last_match_gg_contribution_score',
        category: 'Unknown',
        transform: (value) => value,
    },
    total_kills_m4a1: {
        name: 'M4A1 Kills',
        key: 'm4a1_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_galilar: {
        name: 'Galil-AR Kills',
        key: 'galilar_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_kills_molotov: {
        name: 'Molotov Kills',
        key: 'molotov_kills',
        category: 'Stats',
        transform: (value) => value,
    },
    total_kills_taser: {
        name: 'Taser Kills',
        key: 'taser_kills',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_m4a1: {
        name: 'M4A1 Shots',
        key: 'm4a1_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_galilar: {
        name: 'Galil-AR Shots',
        key: 'galilar_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_shots_taser: {
        name: 'Taser Shots',
        key: 'taser_shots',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_m4a1: {
        name: 'M4A1 Hits',
        key: 'm4a1_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_hits_galilar: {
        name: 'Galil-AR Hits',
        key: 'galilar_hits',
        category: 'Weapons',
        transform: (value) => value,
    },
    total_rounds_map_ar_monastery: {
        name: 'Rounds played in Monaster',
        key: 'ar_monastery_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_matches_won_train: {
        name: 'Matches won in Train ',
        key: 'matches_won_train',
        category: 'Maps_matches', // ???
        transform: (value) => value,
    },
    total_rounds_map_de_vertigo: {
        name: 'Rounds played in Vertigo',
        key: 'de_vertigo_played',
        category: 'Maps',
        transform: (value) => value,
    },
    total_matches_won_baggage: {
        name: 'Matches won in Baggage',
        key: 'matches_won_baggage',
        category: 'Maps_matches',
        transform: (value) => value,
    },
    total_matches_won_lake: {
        name: 'Matches won in Lake',
        key: 'matches_won_lake',
        category: 'Maps_matches',
        transform: (value) => value,
    },
    total_matches_won_stmarc: {
        name: 'Matches won in St Marc',
        key: 'matches_won_stmarc',
        category: 'Maps_matches',
        transform: (value) => value,
    },
    total_matches_won_safehouse: {
        name: 'Matches won in Safehouse',
        key: 'matches_won_safehouse',
        category: 'Maps_matches',
        transform: (value) => value,
    },
    steam_stat_xpearnedgames: {
        name: 'XP-Earned Games',
        key: 'xp_earned_games',
        category: 'Stats',
        transform: (value) => value,
    },
    steam_stat_matchwinscomp: {
        name: 'Matches Won Competitive',
        key: 'steam_stat_matchwinscomp',
        category: 'Maps_matches',
        transform: (value) => value,
    },
}
